### DISCUSSION TITLE
Need help authentification

### DISCUSSION THREAD

**Discussion by innovaweb-dev** [urql-graphql/urql#1705] [Q&A] [answered]
I read documentation about auth, but I need help.
I work on Nuxt project and my server returns a cookie `HtppOnly`

_My questions :_

1.  Where comes `authState` ? It is a React store or a own state of @urql/auth
2. A lot of confusion about the storing of the JWT token, some do not recommend using the `localStroage`
I also read we can copy the token directly in the header for the next requests, but I don't find examples.
Also when a cookie is `HttpOnly`, how we can use it because is not accessible for JavaScript ?

_Ressources :_
https://formidable.com/open-source/urql/docs/advanced/authentication/
https://formidable.com/open-source/urql/docs/api/auth-exchange/#options
https://github.com/FormidableLabs/urql/tree/main/exchanges/auth#quick-start-guide
https://blog.logrocket.com/jwt-authentication-best-practices/
https://www.howtographql.com/react-urql/5-authentication/

**Comment by JoviDeCroock** [comment, marked as answer]
This seems to be a continuation from https://github.com/FormidableLabs/urql/discussions/1694

This is an internal state for the exchange, the storing of the eventual token for cross-session is up to you.
